source/grid/grid_generator.cc

```diff
@@ -5086,19 +5089,25 @@ namespace GridGenerator
                                               Point<dim>(0, 1),  // 2
                                               Point<dim>(1, 1),  // 3
                                               Point<dim>(2, 0),  // 4
-                                              Point<dim>(2, 1)}; // 5
+                                              Point<dim>(2, 1),  // 5
+                                              Point<dim>(3, 0),  // 6
+                                              Point<dim>(3, 1),  // 7
+                                              Point<dim>(1, -1), // 8
+                                              Point<dim>(2, -1), // 9
+                                              Point<dim>(1, 2),  // 10
+                                              Point<dim>(2, 2)}; // 11
 
 
     // consistent orientation
-    unsigned int cell_vertices[n_cells][4] = {{0, 1, 2, 3},  // unit cube
-                                              {1, 4, 3, 5}}; // shifted cube
+    unsigned int cell_vertices[n_cells][4] = {{0, 1, 2, 3},
+                                              {1, 4, 3, 5}, // rotating cube
+                                              {8, 9, 1, 4},
+                                              {4, 6, 5, 7},
+                                              {3, 5, 10, 11}};
 
-    // all 4 true-false combinations of (rotate_left_square | rotate_right_square) to a number 0..3
-    unsigned int this_case = 2 * rotate_left_square + rotate_right_square;
-
-    switch (this_case)
+    switch (n_rotate_middle_square)
       {
-        case /* rotate only right square */ 1:
+        case /* rotate right square */ 1:
```

Comment:
did you mean "rotate the central square"?

Reply:
yes